### PR TITLE
Add valve entity

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Core PR : https://github.com/home-assistant/core/pull/102184

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/valve.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+export function isClosing(stateObj: ValveEntity) {
+  return stateObj.state === "closing";
+}
+
+export function canOpen(stateObj: ValveEntity) {
+  if (stateObj.state === UNAVAILABLE) {
+    return false;
+  }
+  const assumedState = stateObj.attributes.assumed_state === true;
+  return (!isFullyOpen(stateObj) && !isOpening(stateObj)) || assumedState;
```
Comment (bramkragten) [inline src/data/valve.ts:43]:
```suggestion
  return assumedState || (!isFullyOpen(stateObj) && !isOpening(stateObj));
```

File: src/dialogs/more-info/controls/more-info-valve.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  @state() private _mode: Mode = "button";
+
+  private _setMode(ev) {
+    this._mode = ev.currentTarget.mode;
+  }
+
+  protected willUpdate(changedProps: PropertyValues): void {
+    super.willUpdate(changedProps);
+    if (changedProps.has("stateObj") && this.stateObj) {
+      if (!this._mode) {
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-valve.ts:44]: If the `entity_id` changed, I think we should also recalc this?

Comment (piitaya) [reply src/dialogs/more-info/controls/more-info-valve.ts:44]: Yes, we can

**Review by bramkragten — APPROVED** [review]

**Comment by matthiasdebaat** [comment]
Some icon suggestion, in order of my personal preference:
- [`mdi:pipe`](https://pictogrammers.com/library/mdi/icon/pipe/) and [`mdi:pipe-disconnected`](https://pictogrammers.com/library/mdi/icon/pipe-disconnected/)
- [`mdi:water`](https://pictogrammers.com/library/mdi/icon/water/) and [`mdi:water-off`](https://pictogrammers.com/library/mdi/icon/water-off/)
- [`mdi:circle`](https://pictogrammers.com/library/mdi/icon/circle/) and [`mdi:circle-off-outline`](https://pictogrammers.com/library/mdi/icon/circle-off-outline/)

**Pull request merged**